packages/curve-ui-kit/src/shared/ui/SwipeableDrawer.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { Box, styled } from '@mui/material'
+import MuiSwipeableDrawer from '@mui/material/SwipeableDrawer'
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+type Props = {
+  button: React.ReactNode
+  children: React.ReactNode
+  open: boolean
+  setOpen: (open: boolean) => void
+}
+
+const { MaxHeight } = SizesAndSpaces
+/**
+ * The following properties are used in this documentation website for optimal usability of the component:
+ * iOS is hosted on high-end devices. The backdrop transition can be enabled without dropping frames. The performance will be good enough.
+ * iOS has a "swipe to go back" feature that interferes with the discovery feature, so discovery has to be disabled.
+ */
+const iOS = typeof navigator !== 'undefined' && /iPad|iPhone|iPod/.test(navigator.userAgent)
+
+/**
+ * A swipeable drawer component that can be used to display a list of items.
+ * @param param0 button: React.ReactNode, children: React.ReactNode, open: boolean, setOpen: (open: boolean) => void
+ */
+export const SwipeableDrawer = ({ button, children, open, setOpen }: Props) => (
+  <Box sx={{ height: '100%' }}>
+    {button}
+    <MuiSwipeableDrawer
+      disableBackdropTransition={!iOS}
+      disableDiscovery={iOS}
+      anchor="bottom"
+      open={open}
+      onClose={() => setOpen(false)}
+      onOpen={() => setOpen(true)}
+      disableSwipeToOpen={false}
+      keepMounted={false}
+      slotProps={{
+        paper: {
+          sx: (t) => ({
+            backgroundColor: t.design.Layer[1].Fill,
```

Comment:
Should we put these in [`createComponents`](https://github.com/curvefi/curve-frontend/blob/308003c95467dc015848e4a6a28b779b1f9b5a0f/packages/curve-ui-kit/src/themes/components/index.ts#L27)?